app/src/main/java/net/aiscope/gdd_app/ui/metadata/MetadataMapper.kt

```diff
@@ -0,0 +1,80 @@
+package net.aiscope.gdd_app.ui.metadata
+
+import android.content.Context
+import android.content.res.Resources
+import net.aiscope.gdd_app.R
+import net.aiscope.gdd_app.model.MalariaSpecies
+import net.aiscope.gdd_app.model.MalariaStage
+import net.aiscope.gdd_app.model.SmearType
+
+class MetadataMapper {
+    private constructor () { /* make constructor private for utility class */ }
```

Comment:
While I think that this is a very good pattern in Java, in Kotlin doesn't seem to be idiomatic. Since you already have a companion object, a singleton instance will be created, so why don't make the `MetadataMapper` an `object` (thus a singleton)? In this way, the instance of this singleton could be injected into `MetadataActivity`, avoiding the static methods calls (which in general cannot be stubbed in tests) – given that the `@JvmStatic` annotations are removed from the methods. We don't have Java in the codebase, so I would say that this is premature optimization.

Reply:
Thanks for the suggestion! I will make this change.